### ISSUE TITLE
Problems with Handsontable 7 and core-js in ES build

### ISSUE THREAD

**Issue by aaronbeall** [handsontable/handsontable#5888]
### Description

After upgrading to Handsontable 7 our webpack build fails with errors like this:

> ERROR in ./node_modules/handsontable/es/3rdparty/walkontable/src/index.js
> Module not found: Error: Can't resolve 'core-js/modules/web.dom.iterable' in '/Users/abeall/Documents/temp/ht7/node_modules/handsontable/es/3rdparty/walkontable/src'
>  @ ./node_modules/handsontable/es/3rdparty/walkontable/src/index.js 53:0-42
>  @ ./node_modules/handsontable/es/core.js
>  @ ./node_modules/handsontable/es/index.js
>  @ ./src/index.js

In our case we are using `core-js@beta` but you also get this problem with a fresh project that hasn't installed the `core-js` that Handsontable ES build depends on.

### Steps to reproduce

1. Create a fresh repo with `npm init`
2. `npm install webpack webpack-cli handsontable`
3. Add `src/index.js` with this content:
```
import Handsontable from "handsontable"
console.log(Handsontable)
```
4. Run `webpack` (`./node_modules/.bin/webpack`)

Result is a bunch of errors like above: "can't resolve  `core-js/...`" modules.

### Workaround

If you `npm install core-js` it builds without error. However, in my project I already use a different version of `core-js` so I can't just do that. I need to down-grade `core-js` to the version that Handsontable apparently expects.

### Problems

I think there are two related issues here:

1. The ES build has imports for `core-js` but package.json does not declare `core-js` as a dependency. This results in a fresh install having a hidden dependency on a specific version of `core-js` when using the ES version. (The bundled `dist` version doesn't have this problem.)
2. The source code [imports `@babel/polyfill`](https://github.com/handsontable/handsontable/blob/d0aac96b07497347eba7d8307c5b70e0081410f4/src/index.js#L1) which has side-effects. I don't think it should do this. I think the host project is responsible for polyfills. For example in our case, we already use a different version of `core-js`, even if point 1 was resolved it's not clear what the end result would be with `@babel/polyfill` vs our own import of `core-js`, since they aren't the same but both have side-effects on global APIs.

### Fix?

1. Declare `core-js` as a dependency in package.json.
2. [`import "@babel/runtime"`](https://github.com/zloirock/core-js#babelruntime) or import nothing and rely on host app to polyfill for their target browsers. Probably requires some documentation about polyfill requirements, something similar to [React docs](https://reactjs.org/docs/react-dom.html#browser-support).

Thoughts? At the moment we are down-grading to an older version of `core-js` (which requires refactoring code to not use features that were in the newer version) so that we can build with Handsontable 7.

**Comment by robcecil** [comment]
I am blocked using the 7.0.0 version and will instead use 6.2.2 (with React) for the time being.

**Comment by AMBudnik** [comment]
Thank you for reporting the issue.

It is scheduled for March. I hope that we'll be able to fix it soon.

**Comment by wojciechczerniak** [comment]
### TLDR;

To restore webpack support prior `v7.0.0` import Handsontable UMD module:

```js
import Handsontable from `handsontable/dist/handsontable.js`;
```

With wrapper an alias in webpack.conf.js is the only option:

```js
"resolve": {
    "alias": {
        "handsontable": "node_modules/handsontable/dist/handsontable.js"
    }
}
```

### What happened

We've removed `browser` field from `package.json`. Which by default is read by Webpack before `module` and `main` fields. Previous (`browser` field) configuration created a problem where final app had overhead of polyfills and Babel helpers loaded by Handsontable. And I agree with @aaronbeall on this, it's should be the application developer who decide how to handle polyfills.

We didn't liked the idea of `import "@babel/polyfill"` either.

### What can be done

Already mentioned `npm install core-js` fixes the issue for most. Or they already have it installed, or their framework does so they didn't even noticed. We didn't foreseen that someone uses different version of `core-js` already. Sorry for that @aaronbeall @robcecil.

The fix for both is adding `core-js` as a dependency. We're checking this option.

**Comment by aaronbeall** [comment]
FYI core-js@3 is released (out of beta): https://github.com/zloirock/core-js/blob/master/docs/2019-03-19-core-js-3-babel-and-a-look-into-the-future.md

**Comment by wojciechczerniak** [comment]
It was already released few days later when we were fixing the issue 😉 

We're using core-js@3 https://github.com/handsontable/handsontable/pull/5908/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R65

And we have release branch for 7.0.1 already if you want to test it: https://github.com/handsontable/handsontable/tree/release/7.0.1

**Comment by AMBudnik** [comment]
Officially added to Handsontable `7.0.1`
